src/RouteCollection/RouteCollectionUtils.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Dash
+ *
+ * @link      http://github.com/DASPRiD/Dash For the canonical source repository
+ * @copyright 2013-2015 Ben Scholzen 'DASPRiD'
+ * @license   http://opensource.org/licenses/BSD-2-Clause Simplified BSD License
+ */
+
+namespace Dash\RouteCollection;
+
+use Dash\Exception\UnexpectedValueException;
+use Dash\MatchResult\MatchResultInterface;
+use Dash\MatchResult\MethodNotAllowed;
+use Dash\MatchResult\SchemeNotAllowed;
+use Dash\MatchResult\SuccessfulMatch;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Utility methods for working with route collections.
+ */
+final class RouteCollectionUtils
```

Comment:
Also, I think you can move that `matchRouteCollection()` method to `RouteCollection` class